src/ucp/api/ucp.h

```diff
@@ -1500,6 +1528,25 @@ struct ucp_am_recv_param {
      * Endpoint, which can be used for reply to this message.
      */
     ucp_ep_h           reply_ep;
+
+    /**
+     * Length of the whole message in bytes. Relevant for multi-fragment eager
+     * messages handled by data handlers registered without UCP_AM_FLAG_WHOLE_MSG
+     * flag.
+     */
+    size_t             total_length;
+
+    /**
+     * Offset of the message fragment in bytes.
+     */
+    size_t             offset;
```

Comment:
Can we add ascii diagram showing layout of the message and what is offset ?
As for offset - shell we name it offset_to_payload to be more specific in case we will add other types of offset.

Reply:
offset name also has to be more specific please

Reply:
frag_offset?